### PR TITLE
Make PatchOperation<T> public

### PR DESCRIPTION
**Pull request by decodingahmed**
## Description

The `PatchOperation<T>` object is now made public.  This will allow consumers to cast any `PatchOperation` objects to the `PatchOperation<T>` class in order to access the `Value` property. 

For the author's use-case this will allow tests to check that the service level code is passing correct values into the patch request.

In order to avoid introducing a breaking change, these changes do not change the return type of the `Add`, `Set`, `Replace` and `Increment` methods of the `PatchOperation` class.

This PR addresses: https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3374

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Closing issues
Closes https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3374

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs [hunk 1]
```diff
@@ -2,10 +2,11 @@
 // Copyright (c) Microsoft Corporation.  All rights reserved.
 //------------------------------------------------------------
 
-namespace Microsoft.Azure.Cosmos.Tests
+namespace Microsoft.Azure.Cosmos.Tests.Patch
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs:5]: Are these changes needed?
If so, please leave existing conventions (ex: using class prefixed calling etc..)

Comment (decodingahmed) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Patch/PatchOperationTests.cs:5]: Apologies, my IDE automatically applied these changed on "Save". I've added in the class prefixed method calls again.  The rest of the changes I feel are okay.  Please let me know if I've missed any other conventions.
